plugins/scaffolder/api-report.md

```diff
@@ -114,7 +114,7 @@ export class ScaffolderClient implements ScaffolderApi {
   ): Promise<TemplateParameterSchema>;
   // (undocumented)
   listActions(): Promise<ListActionsResponse>;
-  scaffold(templateName: string, values: Record<string, any>): Promise<string>;
+  scaffold(templateName: string, values: Record<string, any>, githubAccessToken: string): Promise<string>;
```

Comment:
So i'm thinking that we need to make this more generic rather than just supporting `github`.

I'm thinking that this parameter should be something like `user` that gets passed in here needs to looks something like this:

```ts
interface User {
  entity: string; // this is the user entity ref in Backstage to a User in the catalog: user:steven or user/test:steven,
  auth?: { // basically for every integration it's a token as an object to allow for flexibility and evolution over time
    github?: { email: string, username: string, token: string },
    bitbucket?: { email: string, username: string, token: string },
    azure?: { email: string, username: string, token: string },
    gitlab?: { email: string, username: string, token: string },
  }
}

We need to know the Backstage user that triggered the the scaffolding and what their corresponding User entity is in Backstage, and then all the auth that they can for every provider they are signed in for.
Then we add this to the context.

Obviously you don't have to implement the other providers, some of them we don't support sign in for anyway, but it's important that we allow room for this API to grow.

We need to check if you're signed in using GithubAuth and then grab the details we need and pass that along instead.

Does that make sense?

Reply:
@OrkoHunter @mtlewis @backstage/maintainers for visibility :eyes:

Reply:
Just realised that actually it's not enough to just specify `github`, we need to also specify the hostname for GHE etc too. So I think it should be more like this:


```ts
interface User {
  entity: string; // this is the user entity ref in Backstage to a User in the catalog: user:steven or user/test:steven,
  auth?: { // basically for every integration it's a token as an object to allow for flexibility and evolution over time
    github?: [{ host: string,  email: string, username: string, token: string }],
    bitbucket?:  [{ host: string,  email: string, username: string, token: string }],
    azure?: [{ host: string,  email: string, username: string, token: string }],
    gitlab?: [{ host: string,  email: string, username: string, token: string }],
  }
}
```

And I think that captures it